### PR TITLE
[Core] Extract all error messages to localizable resources, add codes

### PR DESCRIPTION
**Pull request by paymicro**
### Description of Change ###

Added error codes.
Simplifying PR https://github.com/xamarin/Xamarin.Forms/pull/7353 
- without MEF
- based on the error code from Xamarin.iOS: https://docs.microsoft.com/en-us/xamarin/ios/troubleshooting/mtouch-errors#MT2001
https://github.com/xamarin/xamarin-macios/blob/master/tools/mtouch/error.cs

### Issues Resolved ### 

- fixes [AB#826363](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/826363)
- fixes [AB#1030395](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1030395)
- closes #7353

### API Changes ###
 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Before/After Screenshots ### 

**Before**
Screenshot_4

**After**
image

### Testing Procedure ###

- open any xaml in VisualStudio for Windows
- add `<Button Clicked="BtClick1234"></Button>`
- build project

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jcmanke** [comment]
I would recommend defining the error codes as constants/an enum, so the code isn't littered by magic numbers. `0001` means nothing, `ErrorCodes.CantResolveType` does. It will help future developers properly use existing error codes rather than creating new ones because they miss the fact that an appropriate one has already been defined.

File: Xamarin.Forms.Build.Tasks/CompiledConverters/BindablePropertyConverter.cs [hunk 1]
```diff
@@ -31,42 +32,51 @@ public FieldReference GetBindablePropertyFieldReference(string value, ModuleDefi
...
 
 			var parts = value.Split('.');
-			if (parts.Length == 1) {
+			if (parts.Length == 1)
+			{
 				var parent = node.Parent?.Parent as IElementNode ?? (node.Parent?.Parent as IListNode)?.Parent as IElementNode;
-				if (   (node.Parent as ElementNode)?.XmlType.NamespaceUri == XamlParser.XFUri
-				    && (   (node.Parent as ElementNode)?.XmlType.Name == nameof(Setter)
-				        || (node.Parent as ElementNode)?.XmlType.Name == nameof(PropertyCondition))) {
+				if ((node.Parent as ElementNode)?.XmlType.NamespaceUri == XamlParser.XFUri
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/CompiledConverters/BindablePropertyConverter.cs:38]: that indentation was manually tuned, and perfectly fine

File: Xamarin.Forms.Core/XamlParseExceptionExtension.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Xml;
+
+namespace Xamarin.Forms.Xaml
+{
+	internal static class XamlParseExceptionExtension
+	{
+		internal static IXmlLineInfo GetLineInfo(this IServiceProvider serviceProvider)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/XamlParseExceptionExtension.cs:8]: this is a service provider extension, not a XamlPArseExceptionExtension

File: Xamarin.Forms.Exceptions/XamlParseException.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			var message = string.IsNullOrEmpty(unformattedMessage)
+				? _undefinedMessage
+				: $"{GetTypeString(Error)} {Prefix}{Code:0000}: {SmartFormat(unformattedMessage, args)}";
+
+			return AddPosition(XmlInfo, message);
+		}
+
+		string SmartFormat(string formatString, params object[] args)
+		{
+			var matches = _curlyBracketRegex.Matches(formatString);
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Exceptions/XamlParseException.cs:96]: what's the point of using a regex here ? I think we can rely on string format to do the argument validation

**Pull request closed without merging**